src/index.js

```diff
@@ -69,10 +69,10 @@ ${unused.join(`\n`)}`);
   }
 
   _getFileDepsMap(compilation) {
-    const fileDepsBy = compilation.fileDependencies.reduce((acc, usedFilepath) => ({
-      ...acc,
-      [usedFilepath]: usedFilepath,
-    }), {});
+    const fileDepsBy = compilation.fileDependencies.reduce((acc, usedFilepath) => {
```

Comment:
Mutating reduce. O(n) mutation isn't terrible compared to O(n^2) Object.assign! Though you might get sub milliseconds on native Set usage... haven't tried it.